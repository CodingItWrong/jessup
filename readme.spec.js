const {writeReadme} = require('./readme');

describe('writeReadme', () => {
  describe('cra', () => {
    it('allows returning a readme without cypress', () => {
      expect(writeReadme({framework: 'cra', cypress: false})).toMatchSnapshot();
    });

    it('allows returning a readme with cypress', () => {
      expect(writeReadme({framework: 'cra', cypress: true})).toMatchSnapshot();
    });
  });

  describe('docusaurus', () => {
    it('allows returning a readme without cypress', () => {
      expect(writeReadme({framework: 'doc', cypress: false})).toMatchSnapshot();
    });

    it('allows returning a readme with cypress', () => {
      expect(writeReadme({framework: 'doc', cypress: true})).toMatchSnapshot();
    });
  });

  describe('expo', () => {
    it('allows returning a readme without any testing', () => {
      expect(
        writeReadme({framework: 'expo', unitTesting: false, cypress: false})
      ).toMatchSnapshot();
    });

    it('allows returning a readme with unit testing', () => {
      expect(
        writeReadme({framework: 'expo', unitTesting: true, cypress: false})
      ).toMatchSnapshot();
    });

    it('allows returning a readme with cypress', () => {
      expect(
        writeReadme({framework: 'expo', unitTesting: false, cypress: true})
      ).toMatchSnapshot();
    });

    it('allows returning a readme with both unit testing and cypress', () => {
      expect(
        writeReadme({framework: 'expo', unitTesting: true, cypress: true})
      ).toMatchSnapshot();
    });
  });

  describe('next', () => {
    it('allows returning a readme without cypress', () => {
      expect(
        writeReadme({framework: 'next', cypress: false})
      ).toMatchSnapshot();
    });

    it('allows returning a readme with cypress', () => {
      expect(writeReadme({framework: 'next', cypress: true})).toMatchSnapshot();
    });
  });

  describe('node', () => {
    it('allows returning a readme without unit testing', () => {
      expect(
        writeReadme({framework: 'node', unitTesting: false})
      ).toMatchSnapshot();
    });

    it('allows returning a readme with unit testing', () => {
      expect(
        writeReadme({framework: 'node', unitTesting: true})
      ).toMatchSnapshot();
    });
  });

  describe('node with babel', () => {
    it('allows returning a readme without unit testing', () => {
      expect(
        writeReadme({framework: 'babel', unitTesting: false})
      ).toMatchSnapshot();
    });

    it('allows returning a readme with unit testing', () => {
      expect(
        writeReadme({framework: 'babel', unitTesting: true})
      ).toMatchSnapshot();
    });
  });
});
