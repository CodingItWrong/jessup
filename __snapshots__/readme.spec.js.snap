// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeReadme cra allows returning a readme with cypress 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Running

- Run \`yarn start\`

## Unit Tests

- Run \`yarn test\`

## E2E Tests

- Run the app
- In another terminal, run \`yarn cypress\`
"
`;

exports[`writeReadme cra allows returning a readme without cypress 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Running

- Run \`yarn start\`

## Unit Tests

- Run \`yarn test\`
"
`;

exports[`writeReadme docusaurus allows returning a readme with cypress 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Running

- Run \`yarn start\`

## E2E Tests

- Run the app
- In another terminal, run \`yarn cypress\`
"
`;

exports[`writeReadme docusaurus allows returning a readme without cypress 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Running

- Run \`yarn start\`
"
`;

exports[`writeReadme expo allows returning a readme with both unit testing and cypress 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

Optional:

- To run on Android Emulator, [Android Studio](https://developer.android.com/studio)
- To run on iOS Simulator, [Xcode](https://developer.apple.com/xcode/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Running

- Run \`yarn start\`

## Unit Tests

- Run \`yarn test\`

## E2E Tests

- Run the app
- In another terminal, run \`yarn cypress\`
"
`;

exports[`writeReadme expo allows returning a readme with cypress 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

Optional:

- To run on Android Emulator, [Android Studio](https://developer.android.com/studio)
- To run on iOS Simulator, [Xcode](https://developer.apple.com/xcode/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Running

- Run \`yarn start\`

## E2E Tests

- Run the app
- In another terminal, run \`yarn cypress\`
"
`;

exports[`writeReadme expo allows returning a readme with unit testing 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

Optional:

- To run on Android Emulator, [Android Studio](https://developer.android.com/studio)
- To run on iOS Simulator, [Xcode](https://developer.apple.com/xcode/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Running

- Run \`yarn start\`

## Unit Tests

- Run \`yarn test\`
"
`;

exports[`writeReadme expo allows returning a readme without any testing 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

Optional:

- To run on Android Emulator, [Android Studio](https://developer.android.com/studio)
- To run on iOS Simulator, [Xcode](https://developer.apple.com/xcode/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Running

- Run \`yarn start\`
"
`;

exports[`writeReadme next allows returning a readme with cypress 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Running

- Run \`yarn dev\`

## E2E Tests

- Run the app
- In another terminal, run \`yarn cypress\`
"
`;

exports[`writeReadme next allows returning a readme without cypress 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Running

- Run \`yarn dev\`
"
`;

exports[`writeReadme node allows returning a readme with unit testing 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Unit Tests

- Run \`yarn test\`
"
`;

exports[`writeReadme node allows returning a readme without unit testing 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.
"
`;

exports[`writeReadme node with babel allows returning a readme with unit testing 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.

## Unit Tests

- Run \`yarn test\`
"
`;

exports[`writeReadme node with babel allows returning a readme without unit testing 1`] = `
"# My Project

Describe your project here.

## Requirements

- [Node](https://nodejs.org)
- [Yarn 1.x](https://classic.yarnpkg.com/lang/en/)

## Installation

- Clone the repo
- Run \`yarn install\`

Dependencies are locked with \`yarn.lock\`; please use \`yarn\` rather than \`npm\` for installing.
"
`;
